### PR TITLE
Tweak order logic to calculate total sum instead of amount

### PR DESCRIPTION
**Pull request by wizacass**

### CODE REVIEW

**Review by Zablas — APPROVED** [review]

File: Backend/Models/OrderEntity/Order.cs [hunk 1]
```diff
@@ -66,33 +66,28 @@ public class Order : ISoftDeletable
...
-            Total = CalculateTotalAmount(dto);
+            TotalSum = CalculateTotalAmount(productBalances);
             WarehouseId = dto.WarehouseId;
             PharmacyId = pharmacyId;
             OrderProductBalances = productBalances.Select(pb => new OrderProductBalance(this, pb)).ToList();
         }
 
-        public void UpdateFromDTO(CreateOrderDTO dto, IEnumerable<ProductBalance> productBalances)
+        public void UpdateFromDTO(List<ProductBalance> productBalances)
         {
```
Comment (Anonimas1) [inline Backend/Models/OrderEntity/Order.cs:84]: Why was this changed to List?

**Review by Anonimas1 — APPROVED** [review]

File: Backend/Models/OrderEntity/Order.cs (hunk 1 above)
Comment (wizacass) [reply Backend/Models/OrderEntity/Order.cs:84]: This way we avoid ambiguity of possible double enumartion. Method body has a foreach loop which could be enumerated as both array or a list. If we specify `List` as a parameter instead of `IEnumerable`, whis ambiguity is removed.
